src/main/java/swift/logic/commands/CommandResult.java

```diff
@@ -17,21 +17,34 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    /** The application should switch to contacts tab. */
+    private final boolean isListContact;
+
+    /** The application should switch to tasks tab. */
+    private final boolean isListTask;
+
+    /** The application should display all tasks with associated contacts. */
+    private final boolean isAddTask;
+
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean isListContact,
```

Comment:
I think you need to only have two booleans to keep track of which tab the command result is associated to.